public/script.js

```diff
@@ -6871,6 +6940,11 @@ export async function saveSettings(type) {
         return;
     }
 
+    if (TempResponseLength.isCustomized()) {
+        console.warn('Response length is currently being overridden. Restoring previous value before saving.');
+        TempResponseLength.restore(null);
+    }
```

Comment:
Shouldn't you also switch back to the temp response length after the saving?
I could see this happening with a debounced setting save and then triggering a generation at jus the right time, so the save gets called right after the `genRaw` was called, so the temp length is already set, but the actual generation that needs the response length didn't already happen.

Reply:
That would be a nicer way to handle this.
Worst case it would lead to data loss though - but that *could* already happen with any debounced save.